### ISSUE TITLE
unsupported typescript version 4.2.2 warning 

### ISSUE THREAD

**Issue by paztis** [typescript-eslint/typescript-eslint#3110]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

Typescript 4.2.2 is now released. but estree module is displaying a warning if version is 4.2.0 or upper
Can you fix it ?

```TS

=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.2.0

YOUR TYPESCRIPT VERSION: 4.2.2

Please only submit bug reports when using the officially supported version.

=============
```
**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `4.15.2` |
| `TypeScript`                           | `4.2.2` |
| `node`                                 | `14.15.0` |

**Comment by bradzacher** [comment]
See pinned issue - #2935
We do not yet support v4.2
